File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SmartPhoneNode : public rclcpp::Node
{

public:
    SmartPhoneNode() : Node("robot_news_station")
    {
        subscriber = this->create_subscription<example_interfaces::msg::String>(
            "robot_news", 10,
            std::bind(&SmartPhoneNode::callback, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "subbed to robot news");
    }

private:
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
    void callback(const example_interfaces::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartPhoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class MyNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        self.subscriber = self.create_subscription(
            String, "robot_news", self.callback_robot_news, 10
        )

        self.get_logger().info("SMARTPHONE INITIATED")

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
